riscv/processor.cc

```diff
@@ -942,16 +942,23 @@ void processor_t::register_extension(extension_t* x)
 void processor_t::register_base_instructions()
 {
   #define DECLARE_INSN(name, match, mask) \
-    insn_bits_t name##_match = (match), name##_mask = (mask);
+    insn_bits_t name##_match = (match), name##_mask = (mask); \
+    bool name##_supported = true;
+
   #include "encoding.h"
   #undef DECLARE_INSN
 
+  #define DECLARE_OVERLAP_INSN(name, ext) { name##_supported &= extension_enabled_const(ext); }
```

Comment:
This breaks everything, because this queries `state.misa` to check the writability of the extension bit, and `state.misa` is not initialized until `reset()`, which is called after `register_base_instructions()`.
```
Segmentation fault (core dumped)
```

Instead you should pass in and query `isa` (type `isa_parser_t`) which has been initialized at this point.

Reply:
Sorry. I didn't realize this. Maybe I need use isa->extension_enabled directly. I'll fix it later.